lightning/src/ln/channel.rs

```diff
@@ -3058,6 +3094,33 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before proposing a feerate update, check that we can actually afford the new fee.
+		let inbound_stats = self.get_inbound_pending_htlc_stats(Some(feerate_per_kw));
+		let outbound_stats = self.get_outbound_pending_htlc_stats(Some(feerate_per_kw));
+		let keys = if let Ok(keys) = self.build_holder_transaction_keys(self.cur_holder_commitment_transaction_number) { keys } else { return None; };
+		let commitment_stats = self.build_commitment_transaction(self.cur_holder_commitment_transaction_number, &keys, true, true, logger);
+		let buffer_fee_msat = Channel::<Signer>::commit_tx_fee_sat(feerate_per_kw, commitment_stats.num_nondust_htlcs + CONCURRENT_INBOUND_HTLC_FEE_BUFFER as usize) * 1000;
+		if commitment_stats.local_balance_msat < buffer_fee_msat  + self.counterparty_selected_channel_reserve_satoshis.unwrap() * 1000 {
```

Comment:
Not obvious to me why this doesn't factor in the holding cell HTLC balance (i.e. subtracting it from `local_balance_msat`), like we do in `send_htlc`?

Reply:
Yes, I think we should.

This check enforce that the holder balance is high enough to pay the next commitment transaction fee, at the new proposed feerate. The check(s) in `send_htlc` also enforce that the holder balance is high enough to pay the next commitment transaction fee, even after deducing the sent `amount_msat`. Encompassing to-be-included-in-this-commitment-state-though-not-yet-announced-HTLC (i.e holding cell HTLC balance) give a more accurate answer of the affordance (imo).

Added.